mlflow/java/spark/src/main/scala/org/mlflow/spark/autologging/DatasourceAttributeExtractor.scala

```diff
@@ -109,7 +109,9 @@ object ReplAwareDatasourceAttributeExtractor extends DatasourceAttributeExtracto
         val deltaFileIndexOpt = ReflectionUtils.callMethod(obj, "unapply", Seq(lr)).asInstanceOf[Option[Any]]
         deltaFileIndexOpt.map(fileIndex => {
           val path = ReflectionUtils.getField(fileIndex, "path").toString
-          val versionOpt = Option(ReflectionUtils.callMethod(fileIndex, "tableVersion", Seq.empty)).map(_.toString)
+          val versionOpt = ReflectionUtils.maybeCallMethod(fileIndex, "tableVersion", Seq.empty).orElse(
+            ReflectionUtils.maybeCallMethod(fileIndex, "version", Seq.empty)
+          ).map(_.toString)
```

Comment:
What do you think of this way ?
```
val versionOpt = if (obj.getClass.getName == "com.databricks.sql.transaction.tahoe.files.TahoeBatchFileIndex") {
  None
} else {
  ReflectionUtils.callMethod(...)
}
```

Reply:
I checked runtime code:
https://github.com/databricks/runtime/blob/6ee521683af3f40cd87c199e7ac6c64a4a6dd498/sql/core/src/main/scala/com/databricks/sql/transaction/tahoe/files/TahoeFileIndex.scala#L323

I think we should call `version` method instead ?

Reply:
I saw that too, looks like it was added in DBR 13, should we keep both around just in case?

Reply:
Maybe this way is better:
```
try {
   Some(obj.version)
} catch {
   case _: Exception =>
      Some(obj.tableVersion)
}
```

Reply:
I can try that way as well. If it's able to be done like that, I wonder why the original was done using reflection 🤔

Reply:
Oh, I missed one thing,

we should write:
```

val deltaFileIndex = obj.asInstanceOf[com.databricks.sql.transaction.tahoe.files.TahoeFileIndex]
val versionOpt = try {
   Some(deltaFileIndex.version)
} catch {
   case _: Exception =>
      Some(deltaFileIndex.tableVersion)
}
```

Reply:
Seemingly we can remove another reflection too, like:
```
val obj = ReflectionUtils.getScalaObjectByName("com.databricks.sql.transaction.tahoe.DeltaTable")

obj match {
  case deltaTable: DeltaTable(fileIndex) =>
     val version = try { fileIndex.version } catch { fileIndex.tableVersion }
     ...
  case other => None
}
```

Reply:
Not sure why original code uses reflection, you can reach out to original code author @smurching :)  (Siddharth Murching) , he is in databricks.

Reply:
oh I got it, in OSS we don't have `com.databricks.sql.transaction.tahoe` then importing `DeltaTable` or `TahoeFileIndex` causes the code cannot be compiled in OSS. then seemingly we still have to use reflection.